### ISSUE TITLE
Add support for custom "404 Page not found" handling

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#4037]
## Request

Add functionality to customize the response when a user requests a page that does not exist.

## Motivation

In my experience from work and from running awesome-panel.org it happens that your larger Panel application changes over time. Users might hold on to old links and try to request the page of the old link. What is shown is this

![image](https://user-images.githubusercontent.com/42288570/196857459-e56ef292-5c75-4b39-ad69-05275992c5da.png)

I can see in the logs of awesome-panel.org that users all the time are navigating to pages that do not exist for legacy reasons.

What would be really nice would be to either 1) provide a custom 404 page similar to how you can set the `--index` page or 2) redirect the users to the main entry page.

## Alternatives

What I do at work temporarily is to just keep serving `name_of_old_page.py` file with a message to the user for a period of time. But eventually I would like to remove. And at awesome-panel.org it would not work as it would be confusing to users studying the app to see lots of obsolete `name_of_old_page.py` files.

**Comment by philippjfr** [comment]
Agreed, for now I might suggest that if you're using `pn.serve` you can provide extra_patterns like:

```python
from tornado.web import RedirectHandler

extra_patterns = [
    (r'/old_app', RedirectHandler, {'url': '/new_app'}),
    ...
]

pn.serve(..., extra_patterns=[extra_patterns])
```
